src/pages/result/result.component.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Logo } from '../../shared/logo/logo.component';
+import { DogSearch } from './components/dog-search/dog-search.component';
+import { FilterButton } from './components/filter-button/filter-button.component';
+import { Dog } from './components/dog/dog.component';
+import { fetchPetsAsync } from '../../store/result/actions/result.actions';
+import { RootState } from './result.interfaces';
+import { PetProfile } from '../../shared/interfaces';
+import { FilterValues } from './result.interfaces';
+import { getFiltredPets } from './functions/filter.function';
+
+import './result.styles.scss';
+
+export const Result: React.FC = () => {
+  const [searchedPets, setSearchedPets] = useState<JSX.Element[] | string>('');
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchPetsAsync.request());
+  }, []);
+
+  const pets = useSelector((state: RootState) => state.resultReducer.data);
+
+  // Hardcoden values for filter
+  const filterValues: FilterValues = {
+    carePrice: '1000',
+    price: 'any',
+  };
+
+  const mapArrayOfPets = (petsArray: PetProfile[]): JSX.Element[] => {
+    return petsArray.map(pet => <Dog key={pet._id} name={pet.breed} observations={pet.observations} />);
+  };
+
+  const handleSearchValue = (searchText: string): void => {
+    if (!searchText) setSearchedPets(searchText);
+
+    const filtredPets = pets.filter(pet => pet.breed.toLowerCase().includes(searchText));
+
+    setSearchedPets(mapArrayOfPets(filtredPets));
+  };
+
+  const renderPets = (): string | JSX.Element | JSX.Element[] => {
+    if (searchedPets) {
+      const searchPetsFail = <h4>За вашим запитом нічого не знайдено...</h4>;
+
+      return searchedPets.length ? searchedPets : searchPetsFail;
+    }
+
+    if (Object.keys(filterValues).length !== 0) {
```

Comment:
I'd suggest creating a variable for Object.keys(filterValues).length and also you may simplify your if statement a little bit taking advantage of type-coercion.
I mean here: use if (Object.keys(filterValues).length) instead of if(Object.keys(filterValues).length !== 0).

Reply:
Thaks!